src/Microsoft.PowerShell.ConsoleHost/host/msh/StartTranscriptCmdlet.cs

```diff
@@ -228,7 +227,19 @@ protected override void BeginProcessing()
                         System.IO.File.Delete(effectiveFilePath);
                     }
                 }
-
+                else 
+                {
+                    // check if the path resolves to an existing or non-existing directory. 
+                    string baseDirectory = System.IO.Path.GetDirectoryName(effectiveFilePath);
+                    if(System.IO.Directory.Exists(effectiveFilePath) || (string.Compare(baseDirectory, System.IO.Path.GetFullPath(effectiveFilePath).TrimEnd('\\')) == 0))
```

Comment:
The passed in path could be "c:\foo\\\" or "c:\foo///". And it seems it doesn't matter wither the directory exists or not.
I think maybe you should try other approaches, for example, `Path.GetFileName(path)`. The returned file name would be an empty string if the path passed in is "c:\foo//". (I'm not sure if this approach is perfect, you need to think it through)

And also, please be careful when using `\\` in your path calculation. we are x-plat now, so directory separator can be `//`. Please use `Path.DirectorySeparatorChar` in future.